fuzz/fuzz_targets/descriptor.rs

```diff
@@ -0,0 +1,97 @@
+#![no_main]
+
+use arbitrary::{Arbitrary, Error, Unstructured};
+use gfx_backend_empty::Backend;
+use gfx_descriptor::{DescriptorAllocator, DescriptorCounts};
+use gfx_fuzz::*;
+use hal::{
+    device::Device,
+    pso::{DescriptorSetLayoutBinding, DescriptorType, ImageDescriptorType, ShaderStageFlags},
+};
+
+#[derive(Debug)]
+struct FuzzingInput {
+    bindings: Vec<DescriptorSetLayoutBinding>,
+    repeats: u8,
+    count: u32,
+}
+
+impl Arbitrary for FuzzingInput {
+    fn arbitrary(u: &mut Unstructured) -> Result<Self, Error> {
+        let num_bindings = u.arbitrary_len::<u128>()?;
+        let bindings = (0..num_bindings)
+            .map(|index| arbitrary_binding(index as u32, u))
+            .collect::<Result<Vec<_>, Error>>()?;
+
+        let input = FuzzingInput {
+            bindings,
+            // Allocate the same layout multiple times
+            repeats: u.int_in_range(1..=8)?,
+            // Generate a reasonable count of descriptor sets.
+            count: u.int_in_range(1..=1024)?,
+        };
+        Ok(input)
+    }
+}
+
+// TODO: it would be more elegant to add a new feature to `gfx-backend-empty`,
+// which would generate `Arbitrary` implementations for this type.
+// For now, this is much more convenient.
+fn arbitrary_binding(
+    binding: u32,
+    u: &mut Unstructured,
+) -> Result<DescriptorSetLayoutBinding, Error> {
+    let tys = [
```

Comment:
Can we maybe expose [this](https://github.com/gfx-rs/gfx-extras/blob/33529185c50974983e8e7553ea4841a4e3a5bda2/gfx-descriptor/src/counts.rs#L14) in public, undocumented, so that we can use it here?